torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2323,9 +2323,11 @@ def __init__(self,
                                             **kwargs)
         self.ref = ref
 
-def sample_inputs_foreach(self, device, dtype, N, *, noncontiguous=False):
-    tensors = [make_tensor((N - i, N - i), device, dtype, noncontiguous=noncontiguous) for i in range(N)]
-    return tensors
+def sample_inputs_foreach(self, device, dtype, N, *, noncontiguous=False, same_size=False):
+    if same_size:
+        return [make_tensor((N, N), device, dtype, noncontiguous=noncontiguous) for _ in range(N)]
```

Comment:
This option is used by [`test_binary_op_tensors_on_different_devices`](https://github.com/pytorch/pytorch/pull/59907/files#diff-d11ea193df3e88b33bbb36334bb325f36630ad7c2486057068c58f633a79d1c1R829)